stream_buffer.c

```diff
@@ -478,7 +478,7 @@ static void prvInitialiseNewStreamBuffer( StreamBuffer_t * const pxStreamBuffer,
                                               StaticStreamBuffer_t ** ppxStaticStreamBuffer )
     {
         BaseType_t xReturn;
-        const StreamBuffer_t * const pxStreamBuffer = xStreamBuffer;
+        StreamBuffer_t * const pxStreamBuffer = xStreamBuffer;
```

Comment:
Why do we need to remove const?

Reply:
Clang reports the const qualifier is dropped. Since we returns a non-const pointer, we can remove the const qualifier.
```
/mnt/c/msys64/home/chinglee/kernel/clang/FreeRTOS/FreeRTOS/Source/stream_buffer.c:490:65: error: cast from 'const struct StreamBufferDef_t *' to 'struct xSTATIC_STREAM_BUFFER *' drops const qualifier [-Werror,-Wcast-qual]
            *ppxStaticStreamBuffer = ( StaticStreamBuffer_t * ) pxStreamBuffer;
                                                                ^
1 error generated.
```